cmd/cone/terraform_gen.go

```diff
@@ -0,0 +1,217 @@
+		if err != nil {
+			return err
+		}
+		for _, app := range apps {
+			tmplData := resource.AppTemplate{App: app}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	return nil
+}
+
+func populateResourcesWithPolicies(ctx context.Context, c client.C1Client, object string, resources map[string]resource.TemplateData) error {
+	if object == "policy" || object == "*" {
+		policies, err := c.ListPolicies(ctx)
+		if err != nil {
+			return err
+		}
+		for _, policy := range policies {
+			tmplData := resource.PolicyTemplate{Policy: policy}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	return nil
+}
+
+func populateResourcesWithEntitlements(ctx context.Context, c client.C1Client, v *viper.Viper, object string, resources map[string]resource.TemplateData) error {
+	if object == "app_entitlement" || object == "*" {
+		appId := v.GetString(tfAppIdFlag)
+		if appId == "" {
+			return errors.New("app-id flag is required for app_entitlement object")
+		}
+
+		entitlements, err := c.ListEntitlements(ctx, appId)
+		if err != nil {
+			return err
+		}
+		for _, entitlement := range entitlements {
+			tmplData := resource.AppEntitlementTemplate{AppEntitlement: entitlement}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	return nil
+}
+
+func terraformGen(cmd *cobra.Command, args []string) error {
+	ctx, c, v, err := cmdContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	if err := validateArgLenth(2, args, cmd); err != nil {
+		return err
+	}
+
+	object := args[0]
+	if !slices.Contains(objects, object) && object != "*" {
+		return fmt.Errorf("invalid object name, only support %v and * for all", objects)
+	}
+
+	terraformDir := args[1]
+	if _, err := os.Stat(terraformDir); err != nil {
+		return fmt.Errorf("terraform directory %s does not exist (see here: %s for an example)", terraformDir, terraformProviderExample)
+	}
+
+	// Turns objects into dataTemplates
+	resources, err := getResourceMap(ctx, c, v, object)
+	if err != nil {
+		return err
+	}
+
+	/* TODO @anthony: this all could be simplier if our terraform provider was better at imports.
+	* Currently imports are not supported for some nested objects, for example, policy.steps gets imported incorrectly.
+	* This way of doing it forces datasources to match resources, which is not ideal.
+	*
+	* For each object, we create a template that will import the datasource then output it.
+	 */
+	outputTemplate, err := resource.ApplyTemplates(maps.Values(resources), resource.DataTemplateString, resource.OutputTemplateString)
+	if err != nil {
+		return err
+	}
+	err = writeToFile(terraformDir+"/"+tempTfFile, outputTemplate)
+	if err != nil {
+		return err
+	}
+
+	_, err = os.Create(terraformDir + "/" + tempFile)
+	if err != nil {
+		return err
+	}
+
+	// By running the command the output of terraform plan is piped to a text file
+	pterm.Info.Println("Please run this command in the terraform directory:")
+	pterm.Info.Printfln("terraform plan -no-color > %s", tempFile)
```

Comment:
https://pkg.go.dev/os/exec

Let's not have them run this command themselves, we can execute it programmatically to a temp dir.